### DISCUSSION TITLE
Testing my Premium install

### DISCUSSION THREAD

**Discussion by pauloneill** [bchavez/Bogus#388] [General]
I've just purchased and install Bogus Premium and copied the Licnece file into the root of my dotNet Core application.
(trunk\src\Bookings)

Bogus.dll lives in trunk\src\Bookings\tests\Application.IntegrationTests\bin\Debug\net5.0 but I can't see f => f.Healthcare... for example

Is there a way I can test my installation?

**Comment by bchavez** [comment]
Hi @pauloneill , sorry for the late reply. First off, thank you for the premium subscription purchase and your OSS support.

To clarify, there's no explicit `f => f.Healthcare()` API endpoint. The `Bogus.Healthcare` NuGet package adds the following extension methods to `f => f...`:

https://www.fuget.org/packages/Bogus.Healthcare/33.0.2/lib/netstandard2.0/Bogus.Healthcare.dll/Bogus.Healthcare/ExtensionsForHealthcare

`Bogus.Healthcare`:
* `f => f.Drugs()`
* `f => f.Demographics()`
* `f => f.Human()`
* `f => f.Icd10()`
* `f => f.Icd9()`
* `f => f.Medical()`

See the Bogus README page under "**Bogus Premium Extensions**" here: https://github.com/bchavez/Bogus#bogus-premium-extensions
Eg.
![image](https://user-images.githubusercontent.com/478118/128443106-3eea759d-781f-4092-bcdc-2b083002de0c.png)

So, if you want to generate a random vitamin, the call would be: `f => f.Drugs().Vitamin()`. Hopefully, that clarifies how the Premium APIs tie together.

## Verifying and Testing A Bogus Premium Install

If you'd like to test to ensure your license code works try this:

1. Clone the following repo: https://github.com/bchavez/Bogus.Premium.Test
  Note: The repo contains a test project that should light up all the "bogus premium" features when a valid license code is provided and also includes a sample `Bogus.Healthcare` dataset API call.
2. Replace the `Bogus.Premium.LicenseKey` license file in the cloned repo with your current premium license file.
3. `dotnet build`
4. `dotnet run`

:green_circle: When you have a valid license code, you should get the following output:
```
C:\Code\Projects\Public\Bogus.Premium.Test>dotnet run
Hello Bogus Premium!
{"FirstName":"Valerie","LastName":null,"Age":0,"PainArea":"Lung"}
```

:red_circle: When you have an invalid license code, you should get the following output:
```
C:\Code\Projects\Public\Bogus.Premium.Test>dotnet run
Hello Bogus Premium!
Unhandled exception. System.FormatException: The input is not a valid Base-64 string as it contains a non-base 64 character, more than...
```

Let me know how it works for you!

Thanks,
Brian

**Comment by pauloneill** [comment]
Thanks Brian

Thanks I’ll give this a try – and happy to support something so useful 😃

Thanks for the prompt reply
Best regards,
Paul
p:  (08) 9200 4431
e:  ***@***.******@***.***>
w: www.insilico.com.au<http://www.insilico.com.au/>
Because technology should make life easier…

**Comment by pauloneill** [comment]
Hi Brian

Demo application ran exactly as described – thank you 😃

Now to integrate it into my app…

Cheers

Best regards,
Paul
p:  (08) 9200 4431
e:  ***@***.******@***.***>
w: www.insilico.com.au<http://www.insilico.com.au/>
Because technology should make life easier…
